content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -432,24 +447,27 @@ Since this flow is performing a redemption, you must have both an initiating flo
 Start writing your initiating flow following the same process used when writing the <a href="#write-the-createandissuemarsvoucher-flow">`CreateAndIssueMarsVoucher`</a> flow.
 
 1. Add these annotations: `@InitiatingFlow`, `@StartableByRPC`.
+
 2. Define the `RedeemBoardingTicketWithVoucherInitiator` class with a `@JsonConstructor`, `RpcStartFlowRequestParameters`, and returning a `SignedTransactionDigest`.
 3. Inject these services:
```

Comment:
@peterli-r3 Could you please explain why do I want to add those services and add short description of what each service does?

Reply:
same as my above comment

Reply:
Added descs obtained from @conalsmith-r3